src/bids.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+import pandas as pd
+
+
+def get_bids_version(bids_path):
+    """Returns the BIDS Version of the dataset from the dataset description."""
+    dataset_description = get_dataset_description(bids_path)
+    return dataset_description["BIDSVersion"]
+
+
+def get_dataset_authors(bids_path):
+    """Returns the Authors of the dataset from the dataset description."""
+    dataset_description = get_dataset_description(bids_path)
+    return dataset_description["Authors"]
+
+
+def get_dataset_description(bids_path):
+    """Returns a dictionary with data from the dataset_description json file."""
+    dataset_description_path = bids_path / "dataset_description.json"
+    with open(dataset_description_path, "r") as f:
+        dataset_description = json.load(f)
+    return dataset_description
+
+
+def get_coordinate_frame_description(bids_path, subject_id):
+    """Returns a dictionary with data from the coordsystem json file."""
+    subject_dir = get_subject_dir(bids_path, subject_id)
+    with open(subject_dir / f"{subject_id}_coordsystem.json", "r") as f:
+        return json.load(f)
+
+
+def get_sidecar_metadata(bids_path, subject_id):
+    """Returns a dictionary with data from the subject sidecar json file."""
+    subject_dir = get_subject_dir(bids_path, subject_id)
+    with open(subject_dir / f"{subject_id}_task-tapping_nirs.json", "r") as f:
+        return json.load(f)
+
+
+def get_subject_dir(bids_path, subject_id):
+    """Returns the nirs directory corresponding to the given subject."""
+    return bids_path / subject_id / "nirs"
+
+
+def load_stim_table(bids_path, subject_id):
+    """Returns the table of all stimulus events for the subject."""
+    subject_dir = get_subject_dir(bids_path, subject_id)
+    stim_path = subject_dir / f"{subject_id}_task-tapping_events.tsv"
+    stim_table = pd.read_csv(stim_path, sep="\t")
+    # ignore the start and end of experiment events as they are not stimulus
+    stim_table = stim_table[stim_table.value != 15]
+    return stim_table
+
+
+def list_subject_ids(bids_path):
+    """Returns a list of the ids for the subjects in the dataset."""
+    participant_path = bids_path / "participants.tsv"
```

Comment:
Any idea why I'd get an error when running 
```bash
$ python convert_tapping_dataset_to_nirs.py ~/finger-tapping-fnirs-to-nwb/data/fnirs_finger_tapping
```
where the commanded path is being interpreted by Python as type `bytes` (notice the second line below starts with a b) and therefore the traceback reports `bids_path` is type `bytes`? 
```Python
converting all snirf files in the dataset to nwb
dataset directory: b'/home/ubuntu/finger-tapping-fnirs-to-nwb/data/fnirs_finger_tapping'

Traceback (most recent call last):
  File "convert_tapping_dataset_to_nirs.py", line 139, in <module>
    main()
  File "/home/ubuntu/miniconda3/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/ubuntu/miniconda3/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/ubuntu/miniconda3/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ubuntu/miniconda3/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "convert_tapping_dataset_to_nirs.py", line 60, in main
    for subject_id in list_subject_ids(dataset_path):
  File "src/bids.py", line 57, in list_subject_ids
    participant_path = bids_path / "participants.tsv"
TypeError: unsupported operand type(s) for /: 'bytes' and 'str'
```

Reply:
What version of `click` do you have installed? I forgot to update dependencies, and we need `click~=8.0` because [`click.Path`](https://click.palletsprojects.com/en/8.0.x/api/#click.Path) only returns `pathlib.Path` objects as of 8.0.

You can see in the 2nd like of the output that `dataset_path` is a byte string rather than a `pathlib.Path`.

I updated the requirements file - see if you still have the same error after installing the new requirements.

If you do have the click 8.0+, then we'll have to explore further.

Reply:
Click 8.0 was the issue. This is working properly now.